### DISCUSSION TITLE
Demo scripts for new users?

### DISCUSSION THREAD

**Discussion by Jonk2** [Alex-developer/allskyannotate#3] [General]
I'm glad to have found your annotate software, as I have been writing my own basic python code (very much a beginner) recently to fetch weather data and display as an 'extratext' file.

For example (test setup indoors):

![image](https://user-images.githubusercontent.com/77614536/147709511-65d39628-eac1-4f37-a93c-d7ce79b62fea.png)

As you found out, it was very limting only being able to display basic text, which is why I went for a read and found this!
I've installed it, but Im not entirely sure how to load an example? I usually start with an example and then go and look at the code, pulling it apart until I unsderstand what it's doing.

Do I add my extra text file into this in the "file:" position and change the position to move it about on screen?:

`{
    "fields": {
        "file": {
            "file": "",
            "position": {
                "x": 15,
                "y": 350
            },
            "colour": "white",
            "font": "default"
        }`

If you're interested, I could share my weather data code at a later date, if you fancied incorporating into this?

Thanks

Jon

**Comment by Alex-developer** [comment]
@Jonk2 

Jon

I originally wrote the extra text feature in allsky but quickly realised it was way too limited so have been meaning for a long time to improve it :-)

Yup you got it. Your annotate.json will look like this. Just change the 'ADD THE FULL PATH TO YOUR EXTRA TEXT FILE HERE' to be the full path to your text file. Also don't forget to remove it from the allsky config if its in there otherwise it will be added twice !

```
{
    "fields": {
        "file": {
            "file": "ADD THE FULL PATH TO YOUR EXTRA TEXT FILE HERE",
            "position": {
                "x": 20,
                "y": 420
            },
            "colour": "#666666",
            "font": "default"
        }          
    },
    "images": {    
    },
    "fonts": {
        "moon": {
            "fontPath": "fonts/moon_phases.ttf",
            "fontSize": 36
        },
        "default": {
            "fontPath": "fonts/Roboto-Medium.ttf",
            "fontSize": 36
        }
    }
}
```

If the weather code is obtained by your scripts then you could break it up into several files and add more fields i.e

```
{
    "fields": {
        "file": {
            "file": "ADD THE FULL PATH TO YOUR FIRST EXTRA TEXT FILE HERE",
            "position": {
                "x": 20,
                "y": 420
            },
            "colour": "#666666",
            "font": "default"
        },
        "file1": {
            "file": "ADD THE FULL PATH TO YOUR SECOND EXTRA TEXT FILE HERE",
            "position": {
                "x": 200,
                "y": 0
            },
            "colour": "#666666",
            "font": "default"
        }             
    },
    "images": {    
    },
    "fonts": {
        "moon": {
            "fontPath": "fonts/moon_phases.ttf",
            "fontSize": 36
        },
        "default": {
            "fontPath": "fonts/Roboto-Medium.ttf",
            "fontSize": 36
        }
    }
}
```

Happy to have a look at your weather code. The annotater is designed to have inbuilt fields that people can use. At present it only has Moon data, my local has data for ISS. It would be good to add weather data.

**Comment by Jonk2** [comment]
Thanks for the info - I'm not familiar with json at all....too many {   }! Takes me back to css...anyway.

I've followed your instructions again and with a bit of patience, have the text displayed in different fonts etc. so far so good.

![image](https://user-images.githubusercontent.com/77614536/147840176-8f144163-03a2-4510-af6f-426d91e9e1b6.png)

I've also managed to get an image displayed...but have found what might be 2 issues (not sure if it's me or the code).

1. When I locate an image to the top right hand corner, I would choose 2796,0 as the coords. The chosen compass image is 300px x 300px meaning it should be aligned with the top right hand corner (the main image is 3096 x 2080 px) but doing that, the image is not displayed.
If I pull it in by 1 pixel (2795,1) it displays fine.

I want to put a full sized overlay image with a circular cutout to cover over the parts of the image from the camera I don't want to see, for example the reflections from the lens body, for example:

![image](https://user-images.githubusercontent.com/77614536/147840318-a9b31f6b-0f48-4505-9f5e-32b51d9e6ea1.png)

Setting this to 0,0 should work, but it doesn't.

2. The other thing I'm not sure about, is the compass marks file is a png with transparent background. The annotate.py -v that I'm using to test displays the transparent parts of the png as white (heavily jpegged):

Original image:
 
![image](https://user-images.githubusercontent.com/77614536/147840355-056b5f03-a867-4753-955a-a914f4e6d58d.png)

Overlayed image:
![image](https://user-images.githubusercontent.com/77614536/147840353-64f1861c-d1ba-4514-abc1-23dfe4e17791.png)

Any ideas?

**Comment by Alex-developer** [comment]
I suspect both of those are a bug, for now reduce the images so they are 1 or 2 pixels smaller than the image size and I will look at fixing it

I'll take a look at the image transparency as it does work so could be something up with the compass image


Cheers 

Alex
